src/wallet/test/fuzz/coinselection.cpp

```diff
@@ -217,6 +217,126 @@ FUZZ_TARGET(coin_grinder_is_optimal)
     assert(!result_cg);
 }
 
+FUZZ_TARGET(bnb_finds_min_waste)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+
+    FastRandomContext fast_random_context{ConsumeUInt256(fuzzed_data_provider)};
+    CoinSelectionParams coin_params{fast_random_context};
+    coin_params.m_subtract_fee_outputs = false;
+    // Set effective feerate up to 10'000'000 sats per kvB (10'000 sat/vB).
+    coin_params.m_effective_feerate = CFeeRate{ConsumeMoney(fuzzed_data_provider, 10'000'000), 1'000};
+    coin_params.m_long_term_feerate = CFeeRate{ConsumeMoney(fuzzed_data_provider, 10'000'000), 1'000};
```

Comment:
In a similar test, I run `SelectCoinsBnB` twice and swap fee_rate and long_term_fee_rate, then assert that when long_term_fee_rate is greater less outputs (or equal) are found between the two runs.

Reply:
I did not mean to use two UTXO sets.  If you have a single UTXO set and one swaps the long_term_fee_rate and the short term fee_rate, that ought to produce either the same results or different results depending on if the current fees are expensive or if the current fee_rates are cheap.  I think the test you linked to shows that since the same set is used `{2, 3, 5, 10}` which returns different results depending on the difference in fee_rate and long_term_fee_rate.  The difference between expensive and cheap fee_rates can be toggled by swapping fee_rate for long_term_fee_rate and running BnB again, unless I'm missing something.

Anyway, I understand not wanting to add that complexity to the test.

Reply:
I think we are crossing wires here. I was talking about running coin selection with the **same UTXO set** at two different feerates. You appear to be talking about running a test with two different UTXO sets generated from the same set of _effective values_ at different feerates.

Reply:
> You appear to be talking about running a test with two different UTXO sets generated from the same set of effective values at different feerates. The latter is done already in the test I linked above. For the former the invariant you describe does not hold.

Yes, that is what I'm talking about said a different way.  Yes, the test you linked above tests for that but only for the same static values.